### PR TITLE
Update dns-providers.md

### PR DESCRIPTION
**Pull request by futuredns**
Added provider "FutureDNS", with servers all over the world

### CODE REVIEW

**Review by el-termikael — CHANGES_REQUESTED** [review]
Hello,

Thanks for updating the list of DNS providers. The only thing that needs to be changed is the sdns:// for DNS-over-HTTPS. It is incorrect, as it contains https:// in the host name and a path (/dns-query): https://uploads.adguard.com/el.termikaelcpu05.png

The correct one is sdns://AgcAAAAAAAAAAAAQZG5zLmZ1dHVyZWRucy5tZQA.

Please check how to create and check DNS stamps here: https://github.com/AdguardTeam/KnowledgeBaseDNS/blob/master/README.md#how-to-generate-dnsstamps

**Comment by futuredns** [comment]
I just corrected my mistake. I apologize for the inconvenience.
Thanks for your help

File: docs/general/dns-providers.md [hunk 1]
```diff
@@ -333,6 +333,27 @@ EDNS Client-Subnet is a method that includes components of end-user IP address d
 | DNS, IPv4      | `84.200.69.80` and `84.200.70.40`                  | [Add to AdGuard](sdns://AAAAAAAAAAAADDg0LjIwMC42OS44MA) |
 | DNS, IPv6      | `2001:1608:10:25::1c04:b12f` and `2001:1608:10:25::9249:d69b`            | [Add to AdGuard](sdns://AAAAAAAAAAAAHFsyMDAxOjE2MDg6MTA6MjU6OjFjMDQ6YjEyZl0) |
 
+[FutureDNS](https://futuredns.me/) is a privacy friendly DNS service that blocks ads, trackers, malware and that does not log any data.
+
+|Anycast Network | Address                    |                                                        |
+|----------------|------------------------------------------------------------------|------------------|
+| DNS-over-HTTPS | `https://dns.futuredns.me` | [Add to AdGuard](sdns://AgcAAAAAAAAAAAAQZG5zLmZ1dHVyZWRucy5tZQA)
```
Comment (ameshkov) [inline docs/general/dns-providers.md:340]:
```suggestion
| DNS-over-HTTPS | `https://dns.futuredns.me/dns-query` | [Add to AdGuard](sdns://AgcAAAAAAAAAAAAQZG5zLmZ1dHVyZWRucy5tZQovZG5zLXF1ZXJ5)
```

File: docs/general/dns-providers.md [hunk 2]
```diff
@@ -333,6 +333,27 @@ EDNS Client-Subnet is a method that includes components of end-user IP address d
...
+| DNS-over-TLS   | `tls://dns.futuredns.me` | [Add to AdGuard](sdns://AwcAAAAAAAAAAAAQZG5zLmZ1dHVyZWRucy5tZQ)
+| DNS-over-QUIC  | `quic://dns.futuredns.me` | [Add to AdGuard](sdns://BAcAAAAAAAAAAAAQZG5zLmZ1dHVyZWRucy5tZQ)
```
Comment (ameshkov) [inline docs/general/dns-providers.md:342]: Cannot verify that this one works:
```
% dnslookup example.org quic://dns.futuredns.me    
dnslookup v. 1.7.0
2022/07/05 19:48:45 Cannot make the DNS request: opening quic connection to quic://dns.futuredns.me:853: timeout: no recent network activity
```

```
% dnslookup example.org sdns://BAcAAAAAAAAAAAAQZG5zLmZ1dHVyZWRucy5tZQ
dnslookup v. 1.7.0
2022/07/05 19:49:05 Cannot make the DNS request: opening quic connection to quic://dns.futuredns.me:853: timeout: no recent network activity

```

Please use dnslookup or a similar tool to verify that the addresses (and DNS stamps) are working:
https://github.com/ameshkov/dnslookup

Could it be that the resolver uses a draft version of DoQ?

File: docs/general/dns-providers.md [hunk 3]
```diff
@@ -333,6 +333,27 @@ EDNS Client-Subnet is a method that includes components of end-user IP address d
 | DNS, IPv4      | `84.200.69.80` and `84.200.70.40`                  | [Add to AdGuard](sdns://AAAAAAAAAAAADDg0LjIwMC42OS44MA) |
 | DNS, IPv6      | `2001:1608:10:25::1c04:b12f` and `2001:1608:10:25::9249:d69b`            | [Add to AdGuard](sdns://AAAAAAAAAAAAHFsyMDAxOjE2MDg6MTA6MjU6OjFjMDQ6YjEyZl0) |
 
+[FutureDNS](https://futuredns.me/) is a privacy friendly DNS service that blocks ads, trackers, malware and that does not log any data.
```
Comment (ameshkov) [inline docs/general/dns-providers.md:336]:
```suggestion
### FutureDNS

[FutureDNS](https://futuredns.me/) is a privacy friendly DNS service that blocks ads, trackers, malware and that does not log any data.
```

**Review by ameshkov — CHANGES_REQUESTED** [review]

File: docs/general/dns-providers.md (hunk 2 above)
Comment (ameshkov) [reply docs/general/dns-providers.md:342]: Ah, it uses AdGuard Home according to the description.

@ainar-g have we already published the final DoQ version to the stable branch?

Comment (futuredns) [reply docs/general/dns-providers.md:344]: problem solved
i didn't realize that 853/udp ports were closed
now i tested and everything is working correctly

**Pull request merged**